ocs_ci/ocs/ui/base_ui.py

```diff
@@ -976,6 +988,430 @@ def _retrieve_current_status_from_ui():
             return False
 
 
+class CreateResourceForm(PageNavigator):
```

Comment:
I am curious to know why is this class in base_ui.py? Is this really a base for all UI tests, I don't think so.

Reply:
Same reason why PageNavigator initially is in base_ui.py. CreateResourceForm belongs to multiple resources and the only related parent to it is a PageNavigator

Classically the page classes should be separated by their own files, but python allows to store them all together and some will find it more convenient than to track multiple files across framework.

Reply:
But why not create it somewhere else and inherit `PageNavigator` they way you are doing it here inside base_ui.py. Wouldn't that be ideal?
Existing `PageNavigator` is most commonly used across various tests, but I don't think that the case with `CreateResourceForm`. It's totally on use case. WYDT?

Reply:
@am-agrawa same reason why we use local imports for format_locator - Circular import.
To do so we need to redesign and extract all the classes from base_ui

```
from ocs_ci.ocs.ui.base_ui import PageNavigator
E   ImportError: cannot import name 'PageNavigator' from partially initialized module 'ocs_ci.ocs.ui.base_ui' (most likely due to a circular import) (/Users/danielosypenko/Work/automation_4/ocs-ci/ocs_ci/ocs/ui/base_ui.py)
```
This task will be out of scope of current PR
I wrote myself a ticket https://github.com/red-hat-storage/ocs-ci/issues/7704 I plan complete this task after I finish with feature (I have a close deadline for 2 features)

Reply:
@am-agrawa 
adding clarification -
we can not import anything to base_ui.py and it's BaseUI descendants such as PageNavigator, OverviewTab, etc from outside of the file, because these classes themself inherit from BaseUI. Having them all in one file make "import" not needed

Reply:
I am referring to do something similar to `ocs-ci/ocs_ci/ocs/ui/pvc_ui.py`

Reply:
it is not similar, because all the classes where I would need to import CreateResourceForm are in the base_ui.py hence it will create circular import. Here is the [article](https://agilno.com/blog/why-cyclic-dependency-errors-occur-a-look-into-the-python-import-mechanism/) about it. More about our case here -> https://github.com/red-hat-storage/ocs-ci/issues/7704 Also, the stuck trace I copied above is from my try to accomplish your request.

Reply:
okay!